yt_dlp/extractor/jiosaavn.py

```diff
@@ -36,29 +36,39 @@ class JioSaavnSongIE(JioSaavnBaseIE):
     def _real_extract(self, url):
         audio_id = self._match_id(url)
         song_data = self._extract_initial_data(url, audio_id)['song']['song']
-        media_data = self._download_json(
-            'https://www.jiosaavn.com/api.php', audio_id, data=urlencode_postdata({
-                '__call': 'song.generateAuthToken',
-                '_format': 'json',
-                'bitrate': '128',
-                'url': song_data['encrypted_media_url'],
-            }))
-
-        return {
-            'id': audio_id,
-            'url': media_data['auth_url'],
-            'ext': media_data.get('type'),
-            'vcodec': 'none',
-            **traverse_obj(song_data, {
-                'title': ('title', 'text'),
-                'album': ('album', 'text'),
-                'thumbnail': ('image', 0, {url_or_none}),
-            }),
-        }
+        formats = []
+        # available bitrates are 16, 32, 64, 128, 320
+        for bitrate in self._configuration_arg('bitrate', ['16', '32', '64', '128', '320'], ie_key='JioSaavn'):
```

Comment:
The idea here was that we don't request all bitrates in order to speed up downloading, and instead allow the user to extract additional formats if they so desire. I would also add a check to see if the provided bitrate was valid (the server returns a file, regardless of what you request) and change the order:
```suggestion
        extract_bitrates = self._configuration_arg('bitrate', ['128', '320'], ie_key='JioSaavn')
        if not all(bitrate in ('16', '32', '64', '128', '320') for bitrate in extract_bitrates):
            raise ValueError(f'Invalid bitrate(s) {extract_bitrates}')

        song_data = self._extract_initial_data(url, audio_id)['song']['song']
        formats = []
        for bitrate in extract_bitrates:
```
Please also document this extractor arg in the readme.